keyboards/bastardkb/dilemma/4x6_4/4x6_4.c

```diff
@@ -57,3 +56,58 @@ bool encoder_update_kb(uint8_t index, bool clockwise) {
     return true;
 }
 #endif // ENCODER_ENABLE
+
+#ifdef RGB_MATRIX_ENABLE
+HSV _get_hsv_for_layer_index(uint8_t layer) {
+    switch (layer) {
+        case 1:
+            return (HSV){HSV_BLUE};
+        case 2:
+            return (HSV){HSV_AZURE};
+        case 3:
+            return (HSV){HSV_ORANGE};
+        case 4:
+            return (HSV){HSV_GREEN};
+        case 5:
+            return (HSV){HSV_TEAL};
+        case 6:
+            return (HSV){HSV_PURPLE};
+        case 7:
+        default:
+            return (HSV){HSV_RED};
+    };
+}
+
+// Layer state indicator
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+    if (host_keyboard_led_state().caps_lock) {
+        for (int i = led_min; i <= led_max; i++) {
+            if (HAS_FLAGS(g_led_config.flags[i], LED_FLAG_MODIFIER)) {
+                // set modifier-flagged LEDs to a pure red 76 steps brighter than the current brightness, clamped to 255 (ie. uint8_t max value).
+                rgb_matrix_set_color(i, MIN(rgb_matrix_get_val() + 76, 255), 0, 0);
+            }
+        }
+    }
+
+    uint8_t layer = get_highest_layer(layer_state);
+    if (layer > 0) {
```

Comment:
Add a comment to explain that the per-key LEDs marked as underglow are set to a solid color based on the highest active layer, for any layer apart for the base one. In this case, the default RGB matrix behavior takes precedence. (at least I assume that's what this code does? I don't have a way to double check at the moment.)